provider.go

```diff
@@ -174,20 +168,9 @@ func Provide(c *gin.Context) {
 			}
 		}
 		if task, err := gMT(reqParams.TaskId); err == nil {
-			if len(task.Statuses) == 0 {
-				log.Printf("Rejected token request from %s (Task Id: %s). Reason: %v (no status)", remoteIp, reqParams.TaskId, errTaskNotFresh)
-				atomic.AddInt32(&state.Stats.Denied, 1)
-				c.JSON(403, struct {
-					Status string `json:"status"`
-					Ok     bool   `json:"ok"`
-					Error  string `json:"error"`
-				}{string(state.Status), false, errTaskNotFresh.Error()})
-				return
-			}
 			// https://github.com/apache/mesos/blob/a61074586d778d432ba991701c9c4de9459db897/src/webui/master/static/js/controllers.js#L148
-			startTime := time.Unix(0, int64(task.Statuses[0].Timestamp*1000000000))
-			if time.Now().Sub(startTime) > config.MaxTaskLife {
-				log.Printf("Rejected token request from %s (Task Id: %s). Reason: %v (no status)", remoteIp, reqParams.TaskId, errTaskNotFresh)
+			if len(task.Statuses) > 0 && time.Now().Sub(time.Unix(0, int64(task.Statuses[0].Timestamp*1000000000))) > config.MaxTaskLife {
+				log.Printf("Rejected token request from %s (Task Id: %s). Reason: %v", remoteIp, reqParams.TaskId, errTaskNotFresh)
 				atomic.AddInt32(&state.Stats.Denied, 1)
```

Comment:
Am I missing something here? with this change, if `len(task.Statuses) == 0`, we will issue a token which is wrong.